### ISSUE TITLE
Unison install flow is broken

### ISSUE THREAD

**Issue by jsafoodpanda** [bcpierce00/unison#200]
I've been using unison to sync between mac and a ubuntu server for a while now and everything works as expected.

There seems to have been some changes that make this not work as expected for new users:

`brew install unison` install version 2.51.2
but 
`apt-get install` seems to select 2.48 and that doesn't work as unison requires both versions to be the same.

But now: 
There is no clear way to either install 2.51.2 in ubuntu. Even https://www.seas.upenn.edu/~bcpierce/unison//download/releases/stable/ links to 2.48.2 
And there is no way to brew install unison 2.48 as formulas for those versions seem to be gone.

So, what is the correct setup?

Also, whats up with the GUI? before I could just run unison with the correct configuration file and it would automatically detect changes and sync. Now it doesn't seem to do this anymore.

**Comment by bcpierce00** [comment]
Longer term, the right solution is to get 2.51 into apt-get.  I just posted the official announcement that 2.51 is stable, which may help this happen.  

Short term, your best bet is probably just to recompile from source on the server.  Most people find this pretty straightforward.

Can you give a bit more detail about what's going wrong with the GUI?

**Comment by jsafoodpanda** [comment]
With a bit of research I was able to compile 2.51 on linux.

For my future reference:
 - `apt install ocaml-nox`
 - Download source and cd into folder
 - `make UISTYLE=text`
 - The binary can be found on `src/unison` 
 - `cp src/unison /usr/bin/unison` 

When starting the GUI, first one had to click start (before it would start directly and the option to start a profile automatically did not allow to select any profile).
Second, it doesn't seem to detect changes as fast. I might have to wait a few seconds for the sync to happen or sometimes it didn't seem to happen at all.

**Comment by williscool** [comment]
I second that the install flow is broken.

The only way I could get it work especially with `unison -watch`

was to download an already properly built version

https://medium.com/@yobi.kim/how-to-install-unison-91c9fb36ac09

My steps

```
wget https://github.com/TentativeConvert/Syndicator/raw/master/unison-binaries/unison-2.48.3

wget https://github.com/TentativeConvert/Syndicator/raw/master/unison-binaries/unison-fsmonitor

sudo mv unison-2.48.3 /usr/local/bin/unison
sudo mv unison-fsmonitor /usr/local/bin/unison-fsmonitor

sudo chmod u+x /usr/local/bin/unison
sudo chmod u+x /usr/local/bin/unison-fsmonitor

## 
cd && mkdir .unison && touch .unison/default.prf
```

**Comment by jghub** [comment]
I, too, was struggling today: after upgrade on OSX to 10.13, unison-2.51 is required. our linux-server runs 2.48. result: no more syncing possible between these machines :|. so I followed the steps described above by jsafoodpanda (thanks a lot for these!) and made it work again (after making the new binary be found under the default path in /usr/bin...).

I also would argue for making the command line version w/o GUI the default make target (on all platforms...): I suspect most people want to drive syncing from some personal script or makefile and the GUI just is uncalled for (probably) most of the time, no?

**Comment by mwillema** [comment]
Can we install the latest version of Unison using APT now? Or is there a Docker image of the last version?

**Comment by mgajda** [comment]
I wonder if it is necessary to check server version instead of just protocol version?
Most bug fixes probably does not change the protocol at all.

Temporary solution for Ubuntu might be `apt-get install linuxbrew` then `brew install unison` to get the right version.

**Comment by fenugrec** [comment]
@mgajda , did you have to do anything else to make it work ? on U 18.04, "linuxbrew-wrapper" didn't work properly so I manually installed linuxbrew from their install scripts, but now unison fails with 

```
Unison 2.51.2 (ocaml 4.09.0): Contacting server...
[remote] Shell connection: ssh (ssh, server.org, -e, none, unison, -server)
bash: unison: command not found
```

not sure what's up with that, since I'm setting up the env vars from the remote's ~/.ssh/rc ...

**Comment by erezny** [comment]
I reported the old package to ubuntu so that the maintainer there can update the apt repo.
https://bugs.launchpad.net/ubuntu/+source/unison/+bug/1883030

I've confirmed that unison master can be built and ran manually on ubuntu 20.04, but that's labeled as 2.51.3 and won't work with the 2.51.2 that's in homebrew. compiling on both machines works but I also have to hint where to find the unison binary when connecting to my mac from another computer.

```
sudo apt install ocaml-nox make
git clone https://github.com/bcpierce00/unison.git
cd unison
make UISTYLE=text
~/unison/src/unison -servercmd /usr/local/bin/unison ssh://other/dir ./dir
```
